sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/scenarios/src/batch_stress_tests.cpp

```diff
@@ -0,0 +1,357 @@
+     {"-p", "--partition"},
+     "Partition ID to send events to and receive events from",
+     1},
+    {"MaxTimeouts",
+     {"-m", "--maxtimeouts"},
+     "Number of consecutive receive timeouts allowed before quitting",
+     0},
+    {"UseSasCredential",
+     {"-S", "--useSasCredential"},
+     "Use a SAS credential for authentication",
+     0},
+    {"SleepAfter", {"--sleepAfter"}, "Time to sleep after test completes", 1},
+};
+
+// Default option values.
+
+// Note that the DefaultNumberToSend value is artificially reduced to 100 until the
+// MessageSender::Open code fully supports Open.
+constexpr const std::uint32_t DefaultNumberToSend = /*1000000*/ 100;
+constexpr const std::uint32_t DefaultBatchSize = 1000;
+constexpr const std::uint32_t DefaultPrefetch = 0;
+constexpr const auto DefaultDuration = std::chrono::seconds(60);
+constexpr const std::uint32_t DefaultRounds = 100;
+constexpr const std::uint32_t DefaultPaddingBytes = 1024;
+const std::string DefaultPartitionId{"0"}; // constexpr std::string is a c++17 feature.
+constexpr const std::uint32_t DefaultMaxTimeouts = 10;
+
+const std::vector<EventHubsScenarioOptions>& BatchStressTest::GetScenarioOptions()
+{
+  return BatchScenarioOptions;
+}
+
+void BatchStressTest::Initialize(argagg::parser_results const& parserResults)
+{
+  m_numberToSend = parserResults["NumberToSend"].as<uint32_t>(DefaultNumberToSend);
+  m_batchSize = parserResults["BatchSize"].as<uint32_t>(DefaultBatchSize);
+  m_batchDuration
+      = parserResults["BatchDuration"].as<std::chrono::system_clock::duration>(DefaultDuration);
+  m_prefetchCount = parserResults["Prefetch"].as<uint32_t>(DefaultPrefetch);
+  m_rounds = parserResults["Rounds"].as<uint32_t>(DefaultRounds);
+  m_paddingBytes = parserResults["PaddingBytes"].as<uint32_t>(DefaultPaddingBytes);
+  m_partitionId = parserResults["PartitionId"].as<std::string>(DefaultPartitionId);
+  m_maxTimeouts = parserResults["MaxTimeouts"].as<uint32_t>(DefaultMaxTimeouts);
+  m_verbose = parserResults["verbose"].as<bool>(false);
+  m_useSasCredential = parserResults["UseSasCredential"].as<bool>(true);
+  if (m_rounds == 0xffffffff)
+  {
+    m_rounds = (std::numeric_limits<uint32_t>::max)();
+  }
+
+  auto span{CreateStressSpan("Initialize")};
+  span.first->SetAttribute("NumberToSend", m_numberToSend);
+  span.first->SetAttribute("BatchSize", m_batchSize);
+  span.first->SetAttribute("BatchDuration", m_batchDuration.count());
+  span.first->SetAttribute("Prefetch", m_prefetchCount);
+  span.first->SetAttribute("Rounds", m_rounds);
+  span.first->SetAttribute("PaddingBytes", m_paddingBytes);
+  span.first->SetAttribute("PartitionId", m_partitionId);
+  span.first->SetAttribute("MaxTimeouts", m_maxTimeouts);
+  span.first->SetAttribute("Verbose", m_verbose);
+  span.first->SetAttribute("UseSasCredential", m_useSasCredential);
+
+  //    m_sleepAfter =
+  //    parserResults["SleepAfter"].as<std::chrono::system_clock::duration>(std::chrono::seconds(0));
```

Comment:
Remove commented code

Reply:
That code is commented only because I don't have a good answer for it yet. It will need to come back in a couple of days (once I fix the uAMQP bugs around "Open").